Release notes for 3.5.2 (#13130)
docker/docker.github.io: docker-for-mac/release-notes.md

{% include eula.md %}

> ## Docker Desktop 3.5.2

Reviewer: Is this replacing 3.5.1 with 3.5.2?

Author: Not really, the 3.5.1 release notes are bellow.

Reviewer: Ah it’s the diff getting confused actually…

**Dev Environments Preview**: Dev Environments enable you to seamlessly collaborate with your team members without moving between Git branches to get your code onto your team members' machines. When using Dev Environments, you can share your in-progress work with your team members in just one click, and without having to deal with any merge conflicts. For more information and for instructions on how to use Dev Environments, see [Development Environments Preview](../desktop/dev-environments.md).

### Upgrades

- [Compose V2 beta 6](https://github.com/docker/compose-cli/releases/tag/v2.0.0-beta.6)

>   - `compose run` and `compose exec` use separate streams for stdout and stderr. See [docker/compose-cli#1873](https://github.com/docker/compose-cli/issues/1873).

Reviewer 2: ```suggestion
  - `compose run` and `compose exec` commands now use separate streams for stdout and stderr. See [docker/compose-cli#1873](https://github.com/docker/compose-cli/issues/1873).
```

>   - `compose run` and `compose exec` support detach keys. Fixes [docker/compose-cli#1709](https://github.com/docker/compose-cli/issues/1709).

Reviewer 2: ```suggestion
  - `compose run` and `compose exec` commands support detach keys. Fixes [docker/compose-cli#1709](https://github.com/docker/compose-cli/issues/1709).
```

>   - fixed `--force` and `--volumes` flags on compose rm command. See [docker/compose-cli#1844](https://github.com/docker/compose-cli/issues/1844).

Reviewer 2: ```suggestion
  - Fixed `--force` and `--volumes` flags on `compose rm` command. See [docker/compose-cli#1844](https://github.com/docker/compose-cli/issues/1844).
```

>   - fixed network's IPAM configuration. Service can define a fixed IP. Fixes for [docker/compose-cli#1678](https://github.com/docker/compose-cli/issues/1678) and [docker/compose-cli#1816](https://github.com/docker/compose-cli/issues/1816)

Reviewer 2: ```suggestion
  - Fixed network's IPAM configuration. Service can define a fixed IP. Fixes [docker/compose-cli#1678](https://github.com/docker/compose-cli/issues/1678) and [docker/compose-cli#1816](https://github.com/docker/compose-cli/issues/1816)
```

- Dev Environments

>   - Support VSCode insiders. See [dev-environments#3](https://github.com/docker/dev-environments/issues/3)

Reviewer 2: ```suggestion
  - Support for VS Code Insiders. See [dev-environments#3](https://github.com/docker/dev-environments/issues/3)
```

>   - Allow to specify a branch when cloning. See [dev-environments#11](https://github.com/docker/dev-environments/issues/11)

Reviewer 2: ```suggestion
  - Allow users to specify a branch when cloning a project. See [dev-environments#11](https://github.com/docker/dev-environments/issues/11)
```

### Bug fixes and minor changes

> - Dev Environments: Fix blank screen in some create/remove scenarios. Fixes [dev-environments#4](https://github.com/docker/dev-environments/issues/4)

Reviewer 2: ```suggestion
- Dev Environments: Fixed a blank screen in some create and remove scenarios. Fixes [dev-environments#4](https://github.com/docker/dev-environments/issues/4)
```

> - Dev Environments: Fix error handling in removing an environment. Fixes [dev-environments#8](https://github.com/docker/dev-environments/issues/8)

Reviewer 2: ```suggestion
- Dev Environments: Fixed error handling when removing an environment. Fixes [dev-environments#8](https://github.com/docker/dev-environments/issues/8)
```

> - Dev Environments: Disable Start/Stop/Share buttons while an environment is being created or removed

Reviewer 2: ```suggestion
- Dev Environments: The **Start**, **Stop**, and **Share** buttons are disabled while an environment is being created or removed.
```

> - Fix connection leak when using `virtualization.framework` and not using `vpnkit`.

Reviewer 2: ```suggestion
- Fixed a connection leak when using `virtualization.framework` and not using `vpnkit`.
```

> - Fix spurious traces on iptables updates.

Reviewer 2: ```suggestion
- Fixed spurious traces on iptables updates.
```

> - Fix slowness when adding multiple ports forwarding option.

Reviewer 2: ```suggestion
- Fixed a delay when adding a multiple port forwarding option.
```

### Upgrades

- [Compose V2 beta](https://github.com/docker/compose-cli/releases/tag/v2.0.0-beta.5)
  - `compose run` and `compose exec` use separate streams for stdout and stderr
  - `compose run` and `compose exec` support detach keys
  - fixed `--force` and `--volumes` flags on compose rm command

>   - fixed network's IPAM configuration. Service can define a fixed IP

Reviewer: Should we link to the `compose-cli` issues?

> 

Reviewer 3: - Dev Environments
  - Support VSCode insiders. See https://github.com/docker/dev-environments/issues/3
  - Allow to specify a branche when cloning. See https://github.com/docker/dev-environments/issues/11

Author: Done!

### Bug fixes and minor changes

- Fix connection leak when using `virtualization.framework` and not using `vpnkit`.
- Fix spurious traces on iptables updates.
- Fix slowness when adding multiple ports forwarding option.

> 

Reviewer 3: - Dev Environments: Fix blank screen in some create/remove scenarios. Fixes https://github.com/docker/dev-environments/issues/4
- Dev Environments: Fix error handling in removing an environment. Fixes https://github.com/docker/dev-environments/issues/8
- Dev Environments: Disable Start/Stop/Share buttons while an environment is being created or removed

Author: Added in both mac and win!

## Docker Desktop 3.5.1
2021-06-25

[Download for Mac with Intel chip](https://desktop.docker.com/mac/stable/amd64/66090/Docker.dmg)
[Download for Mac with Apple chip](https://desktop.docker.com/mac/stable/arm64/66090/Docker.dmg)

> ### New

Reviewer 4: We should move the New section to the top, and remove the duplicate Compose v2 beta section.

Author: oh right! I didn't copy everything! Thanks!
